src/modules/article/article.service.ts

```diff
@@ -7,75 +7,104 @@ import { MessageCodeError } from '../../shared/errors/message-code-error';
+      @InjectRepository(User)
+      private readonly userRepository: Repository<User>,
+      @InjectRepository(Category)
+      private readonly categoryRepository: Repository<Category>,
   ) {}
 
   async getAll(): Promise<Article[]> {
-    return this.articleRepository.find();
+    return this.articleRepository.find({ relations: ['categories'] });
   }
 
   async getById(id: string): Promise<Article> {
-    return this.articleRepository.findOne(id);
+    return this.articleRepository.findOne(id, { relations: ['categories'] });
   }
 
   async getArticlesByUserId(userId: string): Promise<Article[]> {
-    return this.articleRepository.find({
-      where: { userId },
-    });
+    return this.articleRepository.find({ where: { userId }, relations: ['categories'] });
   }
 
   async createOne(data: CreateArticleDto): Promise<Article> {
-    const existArticle = await this.articleRepository.findOne({ title: data.title });
-    if (existArticle) throw new MessageCodeError('article:exist');
+    const seoId = ToTranslit(data.title);
+    const article = await this.articleRepository.findOne({
+      where: [{ title: data.title }, { seoId }],
+    });
+    if (article) throw new MessageCodeError('article:exist');
 
-    const existUser = await this.userRepository.findOne({ id: data.userId });
-    if (!existUser) throw new MessageCodeError('user:notFound');
+    const user = await this.userRepository.findOne({ id: data.userId });
+    if (!user) throw new MessageCodeError('user:notFound');
 
-    const existCategory = await this.categoryRepository.findOne({ id: data.categoryId });
-    if (!existCategory) throw new MessageCodeError('category:notFound');
+    const categories = [];
+    let categoryId;
+    for (categoryId of data.categoriesIds) {
+      const category = await this.categoryRepository.findOne({ id: categoryId });
+      if (!category) throw new MessageCodeError('category:notFound');
+      categories.push(category);
+    }
 
     const newArticle = this.articleRepository.create(data);
-    await this.articleRepository.save(data);
-    return newArticle;
+    newArticle.seoId = seoId;
+    newArticle.categories = categories;
+    return this.articleRepository.save(newArticle);
   }
 
   async update(data: UpdateArticleDto): Promise<Article> {
     const article = await this.articleRepository.findOne(data.id);
     if (!article) throw new MessageCodeError('article:notFound');
+    if (data.title) {
+      const seoId = ToTranslit(data.title);
+      const articleExist = !!await this.articleRepository.findOne({ seoId });
+      if (articleExist) throw new MessageCodeError('article:exist');
+      article.seoId = seoId;
+    }
+    if (data.categoriesIds) {
+      let categoryId;
+      for (categoryId of data.categoriesIds) {
+        const existCategory = await this.categoryRepository.findOne({ id: categoryId });
+        if (!existCategory) throw new MessageCodeError('category:notFound');
+      }
```

Comment:
const categories = await this.categoryRepository.find({ id: data.categoriesIds });
if(categories.length!==data.categoriesIds.length) throw new MessageCodeError('category:notFound');

Reply:
done